### ISSUE TITLE
[BUG in 4.8.0] cy.type skips the coma in decimals on input type=number when typing

### ISSUE THREAD

**Issue by sbcomf** [cypress-io/cypress#7775]
**Occurrence: browser with polish language setting (change to reproduce) because input behavior depends on your location/lang settings**

### Current behavior:

**cy.type skips the coma in decimals on input type=number when typing**

When I try to type eg. `2,99` on input cypress types `299` (without coma).
It works with . (dot).

### Desired behavior:

`cy.type` types correct decimal values eg. `2,99; 9,99; 100,9;` (with comas)

### Test code to reproduce
Reproduce on any number input form, but change to eg. polish language of browser (or other uses coma as separator in decimal number format)
```js
it('Test INPUT NUMBER issue', function () {
  cy.viewport(800, 600);
  cy.visit('http://maniax.pl/input-number.html');
  cy.get('[type="number"]').type('9,99').type('{enter}');
});
```

### Versions

Cypress 4.8.0 (newest stable)
latest Chrome, Chromium, Edge
Windows 10 x64
NPM 6.14.5
Enviroment/locale: Polish
Language of webbrowser: polish
![BUG-COMA-INPUT-2020-06-22_14h03_46](https://user-images.githubusercontent.com/48947673/85286422-e9441700-b492-11ea-89f4-bb557bb6b7a9.png)
![BUG-2020-06-22_14h05_38](https://user-images.githubusercontent.com/48947673/85286426-ed703480-b492-11ea-85d5-9e8a5e6af9c2.gif)

**Comment by jennifer-shehane** [comment]
@sbcomf When manually typing in the input provided at http://maniax.pl/input-number.html (outside of Cypress, just open in Chrome), the input **does not type commas** when commas are typed. Cypress is reflecting the exact behavior as a user would encounter on the website. Closing as this is not an issue with Cypress.

**Comment by sbcomf** [comment]
@jennifer-shehane - You don't see commas because you use an English location, when You will change  to eg. polish (or other countries using a comma as separator in the decimal number system (instead of a dot separator). When I visit: http://maniax.pl/input-number.html with polish settings - I can type coma, when I change to english - input accept only dot (as You noticed). So my issue is a high priority bug for countries using a comma as separator.
![2020-06-23_11h08_09](https://user-images.githubusercontent.com/48947673/85386080-6cbd4100-b543-11ea-997c-bfb013f11d6d.png)
![2020-06-23_11h19_51](https://user-images.githubusercontent.com/48947673/85386180-8e1e2d00-b543-11ea-9410-be66c808d7f3.png)

**Comment by roy-dendro** [comment]
Running into same issue.

**Comment by jennifer-shehane** [comment]
Changing my OS's language, I am able to see that the number input allows commas when normally typing and is failing to type the commas in Cypress. 

To change language on MacOS:

- Open **System Preferences**
- Go into **Languages & Region**
- Add a new language "Polish", set to **Primary**
- Restart Chrome
- Visit http://maniax.pl/input-number.html to see commas are type-able.

I'm not sure how we'll be able to test this ourselves. I tried simulating locale in the Chrome browser using `remote:debugger:protocol` and `LANG` env var, but none of that actually changed the behavior in the launched Cypress browser until I set the language in the OS.

**Comment by jennifer-shehane** [comment]
### Repro 

`spec.js`

```js
it('Test INPUT NUMBER issue', () => {
  cy.visit('index.html');
  cy.get('[type="number"]').type('9,99')
  .should('have.value', '999')
  .type('{enter}')
})
```

`index.html`

```html
<html>
<body>
  <h2>Number Field to TEST - Remember: behavior depends on location/language setting in browser</h2>
  <p>The <strong>input type="number"</strong> defines a numeric input field.</p>
  <p>You can use the min and max attributes to add numeric restrictions in the input field:</p>
  <p>Depend on browser settings you have coma or dot separated decimal values</p>
  <form>
    <label for="quantity">Quantity:</label>
    <input type="number" id="quantity" name="quantity" step="any">
    <input type="submit" value="Submit">
  </form>
</body>
</html>
```

### 3.4.1

I am able to see this submit the form in 3.4.1 of cypress, however, it doesn't appear that it was actually typing `9,99` into the field. 

<img width="1328" alt="Zrzut ekranu 2020-07-23 o 5 02 08 PM" src="https://user-images.githubusercontent.com/1271364/88277171-4804e680-cd06-11ea-893f-a7f84fbf5299.png">

### 3.5.0-3.7.0

I do see it typing the comma correctly in the field - the form will not submit on enter however. 

<img width="1332" alt="Zrzut ekranu 2020-07-23 o 5 04 29 PM" src="https://user-images.githubusercontent.com/1271364/88277394-a92cba00-cd06-11ea-8159-a1e22c3c6bf1.png">

### 3.8.0

Doesn't type anything after the first 9 char

<img width="1337" alt="Zrzut ekranu 2020-07-23 o 5 12 49 PM" src="https://user-images.githubusercontent.com/1271364/88278048-c3b36300-cd07-11ea-919a-472436432e02.png">

### 3.8.1

Same behavior as today (4.11.0) where the comma is only number that is excluded. 

<img width="1329" alt="Zrzut ekranu 2020-07-23 o 5 15 01 PM" src="https://user-images.githubusercontent.com/1271364/88278237-112fd000-cd08-11ea-9f90-f37c5bbdb715.png">

---

### Next Steps

So the cause of this issue is this Regex that we use to filter out type-able content into number inputs which was last modified here: https://github.com/cypress-io/cypress/pull/6033 where we explicitly remove typed commas from number inputs to fix this issue: https://github.com/cypress-io/cypress/issues/5997

**As noted in the spec however**

>The formats shown to the user in date, time, and number controls is independent of the format used for form submission.
>https://html.spec.whatwg.org/multipage/input.html#input-impl-notes

As can be seen, even though the comma values are displayed within Chrome in the Poland locale, the value submitted to the form does **not** contain a comma and must be submitted as a floating-point number. 

<img width="517" alt="Zrzut ekranu 2020-07-23 o 5 29 18 PM" src="https://user-images.githubusercontent.com/1271364/88279478-260d6300-cd0a-11ea-88d2-a9198651f925.png">

In conclusion, I would not classify this as a bug in behavior, but rather a visual bug when seeing the input itself. This should not fail any of the behavior of your actual tests and submitting the form, so fixing this will be low priority. 

Please let me know if you have a situation where this does affect the actual behavior of the input/form and we can relook at it. 

This will likely be fixed when we implement Native Events. https://github.com/cypress-io/cypress/issues/311

**Comment by tobmaster** [comment]
We have the same issue and I think this bug is underrated and much more serious. 
Cause while typing in a german locale (where , is the fraction deparator and . is the thousands separator) its not send to the field cause Cypress thinks its invalid as number. But in our locale its perfectly fine. 
Its not only a visual bug, it makes our numeric inputs not working at all when it is floating point number.

If native events really fix it and take time until released there should be a workaround in cypress or a bugfix release that is fixing the issue in keyboard.js. 

Hopefully I this right.

**Comment by theTestingApproach** [comment]
Still facing this issue.

**Comment by testbey** [comment]
Is there a fix for this? I am facing the same issue.

**Comment by edudelta** [comment]
I'm facing same issue. For instance, when trying to type 123,45 the result is 12345.
For now I'm using [cypress-real-events](https://docs.cypress.io/plugins/directory) as workaround.
